src/org/ods/component/UploadToNexusStage.groovy

```diff
@@ -0,0 +1,23 @@
+package org.ods.component
+
+class UploadToNexusStage extends Stage {
+  public final String STAGE_NAME = 'Upload to Nexus'
+
+  UploadToNexusStage(def script, IContext context, Map config) {
+    super(script, context, config)
+  }
+
+  def run() {
+    def repoType = config.repoType ?: "candidates"
+    def distFile = config.distributionFile ?: "${componentId}-${context.tagversion}.tar.gz"
+    def groupId  = config.groupId ?: context.groupId

```

Comment:
Minor style comment: I think it is a little clearer to have these in the constructor to see immediately what config options I have in the code. But we can leave here too ...